pkg/asset/store.go

```diff
@@ -3,8 +3,9 @@ package asset
 // Store is a store for the states of assets.
 type Store interface {
 	// Fetch retrieves the state of the given asset, generating it and its
-	// dependencies if necessary.
-	Fetch(Asset) error
+	// dependencies if necessary. When purging consumed assets, none of the
+	// assets in assetsToPreserve will be purged.
+	Fetch(assetToFetch Asset, assetsToPreserve ...WritableAsset) error
```

Comment:
what was the reason you dropped https://github.com/openshift/installer/pull/1531/files#diff-48efe5e4ccec738eec350b7816cc6ce2R7 in favor of explicit.

do you see that we might want to preserver only part of targets?

Reply:
I am trying to minimize the changes to reduce the contention. The approach here preserves the order in which operations are performed. For each asset in order, there is a fetch of the asset, a save of the state file, a purge of consumed assets, and then a write of the asset files to disk. If all of the assets are passed to `Fetch` as one slice, then none of the assets files would be written to disk until after all of the assets have been fetched.